### DISCUSSION TITLE
execute multiple actions in a single bind

### DISCUSSION THREAD

**Discussion by Fuann-Kinoko** [YaLTeR/niri#212] [Ideas]
usage: 
execute multiple actions, just for convenience

for example
```kdl
 Mod+Shift+Comma  { consume-window-into-column; consume-window-into-column; }
 Mod+Z {
       focus-column-last
       spawn "pkill" "-SIGUSR1" "-o" "waybar"
 }
```

Right now niri only takes the first action within bind, but it is easy to extend it to a Vec and do multiple actions sequentially.

https://github.com/YaLTeR/niri/blob/2b5eeb61620363a325a5c76c4a5d25e45f2a6054/src/input.rs#L1488

However I'm not sure what if multiple actions are trigger in screenshot mode, and some actions are valid like `quit`, while others are not. Should the valid actions be executed and others get ignored?  Or don't do anything if it contains non-screenshot actions?

https://github.com/YaLTeR/niri/blob/2b5eeb61620363a325a5c76c4a5d25e45f2a6054/src/input.rs#L1393-L1405

Additionally, I'm not certain if it's necessary to include a "safety check" to prevent potentially conflicting actions.

If this feature is considered acceptable, I'm willing to submit a pull request.

**Comment by YaLTeR** [comment]
I'm not yet sure how/if I want to allow multiple actions per bind. For now you can work around by binding spawing a script that uses `niri msg action`.
